src/app/clusters/ota-requestor/BDXDownloader.cpp

```diff
@@ -181,7 +193,7 @@ CHIP_ERROR BDXDownloader::HandleBdxEvent(const chip::bdx::TransferSession::Outpu
     case TransferSession::OutputEventType::kBlockReceived: {
         chip::ByteSpan blockData(outEvent.blockdata.Data, outEvent.blockdata.Length);
         ReturnErrorOnFailure(mImageProcessor->ProcessBlock(blockData));
-        mStateDelegate->OnUpdateProgressChanged(mImageProcessor->GetPercentComplete());
+        mStateDelegate->OnUpdateProgressChanged(GetDownloadPercentComplete());
```

Comment:
This assumes linear download. The image processor may only care to look at 30% of the bytes, and employ skipping over parts of sub-images (e.g. an OTA image common across 10 products of a platform, which some elemnts common, some elements product-specific). In that case, only the image processor truly knows the progress, and it's likely based on internal computations of the processor based on image contents.

The download progress is thus a property of the image processor, not the BDX downloader

Reply:
Also, once the image header is actually being parsed (by Image Processor), it should know the entire file size

Reply:
Removed back method to the OTA Image Processor and added incrementing downloaded bytes in nrfconnect platform code.